### ISSUE TITLE
Train multi-document evidence aggregator model (learned)

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#572]
## Problem

Deterministic multi-document extraction (#569) aggregates by counting + scoring, but doesn't model complex evidence relationships:
- Complementary evidence (doc1 has subject, doc2 has predicate)
- Contradictory evidence (different dates in different docs)
- Evidence quality differences (Wikipedia vs forum post)

For complex questions requiring synthesis, a learned aggregator could help.

## Solution

Train a multi-document evidence aggregator (~500K params) using attention over documents.

**Architecture:**
```python
class EvidenceAggregator(nn.Module):
    def __init__(self, compositional_embedding):
        self.comp_emb = compositional_embedding  # Frozen 128D
        
        # Document encoder
        self.doc_encoder = nn.GRU(128, 128, batch_first=True)
        
        # Cross-attention: query attending to documents
        self.attention = nn.MultiheadAttention(128, num_heads=4)
        
        # Evidence scorer
        self.scorer = nn.Sequential(
            nn.Linear(128 * 2, 256),  # query + attended docs
            nn.ReLU(),
            nn.Linear(256, 1),
            nn.Sigmoid()
        )
    
    def forward(self, query_emb, doc_embs):
        # Encode documents
        doc_hiddens, _ = self.doc_encoder(doc_embs)  # [batch, num_docs, 128]
        
        # Query attends to documents
        attended, weights = self.attention(
            query_emb.unsqueeze(0),  # query
            doc_hiddens,  # keys
            doc_hiddens   # values
        )
        
        # Score evidence quality
        combined = torch.cat([query_emb, attended.squeeze(0)], dim=-1)
        return self.scorer(combined), weights
```

**Training data:**
- Positive: Questions where top-3 docs contain answer
- Negative: Questions where top-3 docs don't contain answer
- Hard negatives: Mixed (some relevant, some not)

## Expected Impact

+10-15% on complex questions requiring multi-document reasoning.

## Dependencies

- Depends on #569 (deterministic multi-doc extraction baseline)
- Optional: Compare learned vs deterministic to justify parameters

## When to Implement

**Priority P2** - Do this AFTER deterministic improvements show limits.

1. Implement #569 (deterministic aggregation) first
2. Evaluate: Does deterministic work well enough?
3. If deterministic aggregation still fails on complex cases, then train this model

## Implementation

1. Generate training data from evaluation results
2. Train aggregator model (~4 hours GPU)
3. Compare with deterministic baseline (#569)
4. If significant improvement, integrate into pipeline

**Estimated time:** 1 week
**Params:** ~500K (total system: 320K + 180K + 500K = 1M)

**Comment by marctjones** [comment]
Closing as part of a 2026-09-08 backlog cleanup (see PLAN.md, DESIGN.md, CLAUDE.md's merge gate).

This issue predates the merge gate (adopted 2026-07-13) and the AST-native orchestrator rewrite. It describes work against the pre-orchestrator RAG pipeline or the learned model stack (M1/M2 embeddings, compositional embeddings, entity classifiers, neural rerankers) that was pruned from the repo -- klareco/embeddings/, klareco/models/, and klareco/summarization/ no longer exist. It has no milestone and was never benchmarked, matching exactly the failure mode CLAUDE.md documents (25 capability issues filed 2026-05-26, 8 implemented, none benchmarked).

Labeled archived-2026-09 so this batch is recoverable: gh issue list --state closed --label archived-2026-09. If this capability is still wanted, re-file it against the current architecture naming the benchmark metric and test the merge gate requires.
